emcc.py

```diff
@@ -1442,14 +1442,18 @@ def default_setting(name, new_default):
         settings.EXPORT_ALL = 1
       settings.RELOCATABLE = 1
 
+    if settings.MAIN_MODULE:
+      settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += [
+          '$getDylinkMetadata',
+          '__heap_base',
+          '__stack_pointer',
+      ]
```

Comment:
why are these only needed in a main module?

Reply:
`getDylinkMetadata` (in `library_dylink.py`) is only defined when `MAIN_MODULE` is defined.  Without this change I see `$getDylinkMetadata` as undefined when building with `RELOCATABLE`.

The other two I was able to move back to the original location.

Reply:
> > Also, allow programs built with -sLINKABLE to report undefined symbol errors. By default they don't (but this can be overridden on the command line).
> 
> Where is that default set?

Ah I think I meant `RELOCATABLE`:

```
if settings.RELOCATABLE:                                                                         
      default_setting('ERROR_ON_UNDEFINED_SYMBOLS', 0)                                                          
      default_setting('WARN_ON_UNDEFINED_SYMBOLS', 0)    
```

I guess I need to update that block to include LINKABLE too.

Reply:
Done